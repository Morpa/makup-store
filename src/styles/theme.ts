export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },

  border: {
    radius: '0.4rem'
  },

  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans- serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  },

  colors: {
    background: '#141620',
    shape: '#090B10',
    gradient:
      'linear-gradient(90deg, rgba(0,53,157,1) 5%, rgba(0,111,255,1) 50%, rgba(63,204,255,1) 100%)',
    primary: 'rgba(0,53,157,1)',
    secondary: '#D6AE2F',
    border: '#212b54',
    hover: '#212b54',
    title: '#BFC4D3',
    subtitle: '#9e9e9e',
    subtext: 'rgb(88, 88, 88)'
  },

  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },

  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const
