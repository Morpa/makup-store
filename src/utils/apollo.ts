import { ApolloClient, HttpLink, NormalizedCacheObject } from '@apollo/client'
import { useMemo } from 'react'

import apolloCache from './apolloCache'

let apolloClient: ApolloClient<NormalizedCacheObject>

function createApolloClient() {
  const httpLink = new HttpLink({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql/`
  })

  return new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: httpLink,
    cache: apolloCache
  })
}

export function initializeApollo(initialState = {}) {
  const apolloClientGlobal = apolloClient ?? createApolloClient()

  if (initialState) {
    apolloClientGlobal.cache.restore(initialState)
  }

  if (typeof window === 'undefined') return apolloClientGlobal

  apolloClient = apolloClient ?? apolloClientGlobal

  return apolloClient
}

export function useApollo(initialState = {}) {
  const store = useMemo(() => initializeApollo(initialState), [initialState])
  return store
}
