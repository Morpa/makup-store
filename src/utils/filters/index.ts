import { ParsedUrlQueryInput } from 'querystring'

import { ItemProps } from 'components/ExploreSidebar'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
  filterItems: Pick<ItemProps, 'type' | 'name'>[]
}

export const parseQueryStringToFilter = ({
  queryString,
  filterItems
}: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  Object.keys(queryString).forEach((key) => {
    const item = filterItems?.find((item) => item.name === key)
    const isCheckbox = item?.type === 'checkbox'
    const isArray = Array.isArray(queryString[key])

    obj[key] = !isArray && isCheckbox ? [queryString[key]] : queryString[key]
  })

  return obj
}
