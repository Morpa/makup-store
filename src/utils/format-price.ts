export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('pt-PT', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
