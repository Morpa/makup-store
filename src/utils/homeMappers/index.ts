import { ProductApiResponse } from 'services/api'

export const productsMapper = (products: ProductApiResponse[]) => {
  return products
    ? products.map((item) => {
        return {
          id: item.id,
          title: item.name,
          brand: item.brand?.toLocaleUpperCase(),
          img: `https:${item.api_featured_image}`,
          price:
            Number(item.price) > 0
              ? item.price
              : Math.floor(Math.random() * 50),
          rating: Number(item?.rating) > 0 ? Number(item.rating) : Number(0)
        }
      })
    : {}
}

export const bannersMapper = (products: ProductApiResponse[]) => {
  return products
    ? products.map((item) => {
        return {
          img: `https:${item.api_featured_image}`,
          title: item.name,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`
        }
      })
    : {}
}

export const highlightMapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  products: any,
  title: string,
  subtitle: string,
  alignment: string
) => {
  return products
    ? products.map((item: ProductApiResponse) => {
        return {
          title,
          subtitle,
          backgroundImage: `https:${item.api_featured_image}`,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`,
          alignment
        }
      })
    : {}
}
