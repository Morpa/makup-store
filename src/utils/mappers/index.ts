import { QueryProducts_products_product } from 'graphql/generated/QueryProducts'

export const productsMapper = (products: QueryProducts_products_product[]) => {
  return products
    ? products.map((item) => {
        return {
          id: item.id,
          title: item.name,
          brand: (item.brand?.length && item.brand?.toLocaleUpperCase()) || '',
          img: `https:${item.api_featured_image}`,
          price:
            Number(item.price) > 0
              ? item.price
              : Math.floor(Math.random() * 50),
          rating: Number(item?.rating) > 0 ? Number(item.rating) : Number(0)
        }
      })
    : {}
}

export const bannersMapper = (products: QueryProducts_products_product[]) => {
  return products
    ? products.map((item) => {
        return {
          img: `https:${item.api_featured_image}`,
          title: item.name,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`
        }
      })
    : {}
}

export const highlightMapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  products: any,
  title: string,
  subtitle: string,
  alignment: string
) => {
  return products
    ? products.map((item: QueryProducts_products_product) => {
        return {
          title,
          subtitle,
          backgroundImage: `https:${item.api_featured_image}`,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`,
          alignment
        }
      })
    : {}
}
