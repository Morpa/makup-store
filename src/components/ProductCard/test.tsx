import { render, screen } from 'utils/test-utils'

import theme from 'styles/theme'

import { ProductCard } from '.'

const props = {
  id: '1',
  title: 'Lippie Pencil',
  brand: 'Colourpop',
  img: 'https://source.unsplash.com/user/willianjusten/300x140',
  price: '5.0',
  rating: '4'
}

describe('<ProductCard />', () => {
  it('should render correctly', () => {
    const { container } = render(<ProductCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: props.brand })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.img
    )

    expect(screen.getByRole('link', { name: props.title })).toHaveAttribute(
      'href',
      `/product/${props.id}`
    )

    expect(screen.getByLabelText(/add to cart/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render price in label', () => {
    render(<ProductCard {...props} />)

    const price = screen.getByText('5,00 â‚¬')

    expect(price).not.toHaveStyle({ textDecoration: 'line-throug' })

    expect(price).toHaveStyle({ backgroundColor: theme.colors.secondary })
  })
})
