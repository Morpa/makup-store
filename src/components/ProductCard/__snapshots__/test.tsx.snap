// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render correctly 1`] = `
.c10 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
}

.c9:focus {
  outline: 1px dashed;
}

.c9:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c9 svg {
  width: 1.5rem;
}

.c9 svg + span {
  margin-left: 0.8rem;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #FAFAFA;
}

.c1 {
  min-height: 14rem;
  position: relative;
  width: 100%;
  background: #f6f7f8;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  height: 100%;
  margin: 1.6rem;
}

.c3 {
  max-width: calc(70% - 2.5rem);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-weight: 600;
  color: #030517;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #8F8F8F;
}

.c6 {
  color: #F231A5;
  position: absolute;
  right: -1rem;
  top: -0.5rem;
}

.c6 svg {
  width: 2.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

<article
  class="c0"
>
  <a
    class="c1"
    href="/product/1"
  >
    <img
      alt="Lippie Pencil"
      layout="fill"
      src="https://source.unsplash.com/user/willianjusten/300x140"
    />
  </a>
  <div
    class="c2"
  >
    <a
      class="c3"
      href="/product/1"
    >
      <h3
        class="c4"
      >
        Lippie Pencil
      </h3>
      <h4
        class="c5"
      >
        Colourpop
      </h4>
    </a>
    <div
      class="c6"
    >
      <span
        class=""
      >
        <span
          style="color: rgb(241, 165, 69); width: 25px; height: 25px; display: inline-flex;"
        >
          <svg
            fill="currentColor"
            height="25"
            viewBox="0 0 24 24"
            width="25"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          style="color: rgb(241, 165, 69); width: 25px; height: 25px; display: inline-flex;"
        >
          <svg
            fill="currentColor"
            height="25"
            viewBox="0 0 24 24"
            width="25"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          style="color: rgb(241, 165, 69); width: 25px; height: 25px; display: inline-flex;"
        >
          <svg
            fill="currentColor"
            height="25"
            viewBox="0 0 24 24"
            width="25"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          style="color: rgb(241, 165, 69); width: 25px; height: 25px; display: inline-flex;"
        >
          <svg
            fill="currentColor"
            height="25"
            viewBox="0 0 24 24"
            width="25"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          style="color: rgb(204, 204, 204); width: 25px; height: 25px; display: inline-flex;"
        >
          <svg
            fill="currentColor"
            height="25"
            viewBox="0 0 24 24"
            width="25"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        5,00 €
      </div>
      <button
        aria-label="Add to cart"
        class="c9"
      >
        <svg
          aria-hidden="true"
          class="c10"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
          />
        </svg>
      </button>
    </div>
  </div>
</article>
`;
